### PR TITLE
MAINT: Add a Dependabot config to auto-update GitHub action versions

### PR DESCRIPTION
**Pull request by kurtmckee**
#### Reference issue
N/A

#### What does this implement/fix?
This PR introduces a Dependabot configuration that will submit a single PR on a monthly basis to update GitHub action versions. The PR will batch all available action updates together.

If this merges, you can anticipate a single PR to be opened immediately to update e.g. `actions/checkout@v3` to `v4`.

#### Additional information
No additional info. I think that I followed the pre-PR checklist and followed the contributing guidelines, but please let me know if there's anything that needs to be addressed (for example, if I need to mark the commit as signed-off-by or something like that). Thanks!

### CODE REVIEW

File: .github/dependabot.yml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+version: 2
+updates:
+  - package-ecosystem: "github-actions"
+    directory: "/"
+    schedule:
+      interval: "monthly"
+    groups:
```
Comment (math411) [inline .github/dependabot.yml:7]:
```suggestion
    commit-message:
      prefix: MAINT
    groups:
```

Comment (kurtmckee) [reply .github/dependabot.yml:7]: Goodness gracious, I overlooked the commit message section of the contributing guide and only looked at the checklist. I'll update the commit and PR. Thanks for catching this!

Comment (kurtmckee) [reply .github/dependabot.yml:9]: I think this is addressed now. Thanks!

**Comment by lucascolley** [comment]
This was suggested in gh-17327 and decided against there. Perhaps folks have changed their mind but I'd imagine this is still unwanted unless things have improved. See in particular https://github.com/scipy/scipy/pull/17327#issuecomment-1303495335

**Comment by rgommers** [comment]
Still unwanted I'm afraid. Thanks for the attempt to contribute though @kurtmckee, much appreciated.

**Comment by kurtmckee** [comment]
No worries. I'm aware of that past behavior and associated discussion but it has since been addressed.

Have a great weekend!

**Pull request closed without merging**